src/src/com/microsoft/aad/adal/DateTimeAdapter.java

```diff
@@ -67,12 +86,22 @@ public synchronized Date deserialize(JsonElement json, Type typeOfT,
 
         try {
             return localFormat.parse(jsonString);
-        } catch (ParseException ignored) {
+        } catch (final ParseException ignored) {
+        }
+        
+        try {
+            return local24HourFormat.parse(jsonString);
+        } catch (final ParseException ignored) {
         }
 
         try {
             return enUsFormat.parse(jsonString);
-        } catch (ParseException ignored) {
+        } catch (final ParseException ignored) {
+        }
+
+        try {
+            return enUs24HourFormat.parse(jsonString);
+        } catch (final ParseException ignored) {
         }
 
         try {
```

Comment:
please change the order and start with iso8601Format, because it should be the most popular